pkg/quay/quay.go

```diff
@@ -401,3 +401,66 @@ func handleRobotName(robotName string) (string, error) {
 	}
 	return robotName, nil
 }
+
+func (c *QuayClient) GetTagsFromPage(organization, repository string, page int) ([]Tag, bool, error) {
+	url := fmt.Sprintf("%s/repository/%s/%s/tag/?page=%d", c.url, organization, repository, page)
+
+	req, err := http.NewRequest(http.MethodGet, url, nil)
+	if err != nil {
+		return nil, false, fmt.Errorf("failed to create new request, error: %s", err)
+	}
+
+	req.Header.Add("Authorization", fmt.Sprintf("Bearer %s", c.AuthToken))
+	req.Header.Add("Content-Type", "application/json")
+
+	res, err := c.httpClient.Do(req)
+	if err != nil {
+		return nil, false, fmt.Errorf("failed to Do request, error: %s", err)
+	}
+	if res.StatusCode != 200 {
+		return nil, false, fmt.Errorf("error getting repositories, got status code %d", res.StatusCode)
+	}
+
+	defer res.Body.Close()
+	body, err := io.ReadAll(res.Body)
+	if err != nil {
+		return nil, false, fmt.Errorf("failed to read response body, error: %s", err)
+	}
+
+	var response struct {
+		Tags          []Tag
+		Page          int
+		HasAdditional bool
+	}
+	err = json.Unmarshal(body, &response)
+	if err != nil {
+		return nil, false, fmt.Errorf("failed to unmarshal body, error: %s", err)
+	}
+	return response.Tags, response.HasAdditional, nil
+}
+
+func (c *QuayClient) DeleteTag(organization, repository, tag string) (bool, error) {
+	url := fmt.Sprintf("%s/repository/%s/%s/tag/%s", c.url, organization, repository, tag)
+
+	req, err := http.NewRequest(http.MethodDelete, url, nil)
+	if err != nil {
+		return false, err
+	}
+
+	req.Header.Add("Authorization", fmt.Sprintf("%s %s", "Bearer", c.AuthToken))
+	req.Header.Add("Content-Type", "application/json")
+
+	res, err := c.httpClient.Do(req)
+	if err != nil {
+		return false, fmt.Errorf("failed to Do request, error: %s", err)
+	}
+	defer res.Body.Close()
+
+	if res.StatusCode == 204 {
+		return true, nil
+	}
+	if res.StatusCode == 404 {
+		return false, nil
+	}
```

Comment:
The final result of deleting a tag is the tag does not exist any more. When Quay.io responds 404, isn't it a kind of expectation? Or, 404 has other special meaning in this context?

Reply:
I see your point, I think it might be a matter of personal preference. 
My reasons for leaving it in are:
- I believe it is more flexible, off the top of my head, I can imagine wanting to log if the tag was actually deleted by the function or if it was not found.
- Its behavior is consistent with DeleteRepository and DeleteRobotAccounts functions, so those would have to be changed as well.